### ISSUE TITLE
Inputs to the CORAL loss

### ISSUE THREAD

**Issue by cmhungsteve** [SSARCandy/DeepCORAL#4]
out1, out2 = model(source_data, target_data)

        classification_loss = torch.nn.functional.cross_entropy(out1, source_label)
        coral = models.CORAL()
        coral_loss = coral(out1, out2)

        sum_loss = _lambda*coral_loss + classification_loss
From your codes, the input to the CORAL loss function is the prediction. Although it seems the case in the original DeepCORAL paper, it's not really reasonable to me. To my understanding, the main goal of domain adaptation is to match the feature distribution between source and target domain, so I think it makes more sense to calculate the loss function using the feature vectors from previous layers (e.g. fc6, fc7). 
In addition, in some famous DA works:
https://arxiv.org/pdf/1412.3474.pdf
http://proceedings.mlr.press/v37/ganin15.pdf
https://arxiv.org/pdf/1502.02791.pdf
They all have loss functions for those layers. Maybe you can try different positions of the CORAL loss function and see whether it improve the performance or not.
Thanks.

**Comment by SSARCandy** [comment]
Thanks for your advise!

I tried using `fc7` as CORAL loss input, and the accuracy on test case is ~49%.
that is slightly lower than using `fc8` as input.

But I agree your opinion, using feature vectors from previous layers may makes more sense.
I'll keep trying using other layer as input, thanks for your help!

**Comment by cmhungsteve** [comment]
No problem. I haven't really implemented any DA algorithm. I believe there are lots of tricks for implementation. The correct idea doesn't correspond to good performance. Good luck and hope you can get better results.

**Comment by cmhungsteve** [comment]
Is there any update?
